cmd/compose/port.go

```diff
@@ -62,11 +68,40 @@ func portCommand(p *ProjectOptions, dockerCli command.Cli, backend api.Service)
 	return cmd
 }
 
+func printPorts(dockerCli command.Cli, publishers api.PortPublishers, opts portOptions) {
+	for _, publisher := range publishers {
+		if publisher.Protocol != opts.protocol {
+			continue
+		}
+
+		var format string
+		if strings.Contains(publisher.URL, ":") {
+			format = "%d/%s -> [%s]:%d\n"
+		} else {
+			format = "%d/%s -> %s:%d\n"
+		}
+		fmt.Fprintf(dockerCli.Out(), format, publisher.PublishedPort, publisher.Protocol, publisher.URL, publisher.TargetPort)
+	}
+}
+
 func runPort(ctx context.Context, dockerCli command.Cli, backend api.Service, opts portOptions, service string) error {
 	projectName, err := opts.toProjectName(ctx, dockerCli)
 	if err != nil {
 		return err
 	}
+
+	if opts.list {
+		summaries, err := backend.Ps(ctx, projectName, api.PsOptions{Services: []string{service}})
+		if err != nil {
+			return err
+		}
+		if len(summaries) <= opts.index {
```

Comment:
[]ContainerSummary index is unrelated to replica index in service set by `--index`.